### ISSUE TITLE
Add Windows ARM64 support

### ISSUE THREAD

**Issue by dennisameling** [Microsoft/vscode-filewatcher-windows#18]
VS Code [is available for Windows ARM64 natively](https://code.visualstudio.com/updates/v1_46#_windows-arm64-insiders), which results in great performance. However, `CodeHelper.exe` is still x86, which uses emulation on Windows ARM64, resulting in slower performance.

It'd be great if `CodeHelper.exe` could be made available for Windows ARM64 natively. Currently, this module [targets .NET Framework 4.5](https://github.com/microsoft/vscode-filewatcher-windows/blob/main/FileWatcher/FileWatcher.csproj#L12), which doesn't support ARM64. Therefore, it would first need to be upgraded to .NET 5.

I'm happy to help with upgrading if the team would like that, and I have some first work done in https://github.com/dennisameling/vscode-filewatcher-windows/tree/dotnet-5, which compiles just fine. The team would need to decide which deployment model they prefer ([self-contained or framework-dependent](https://docs.microsoft.com/en-us/dotnet/core/deploying)), which may have a significant impact on binary size.

CC @joaomoreno who mentioned the wish for native ARM64 binaries in https://github.com/microsoft/vscode/issues/98587

**Comment by joaomoreno** [comment]
Yeah this would be cool to have and likely easy as well.

**Comment by bpasero** [comment]
Yeah maybe, maybe not. I have had (and still have) my doubts that shipping 3 different file watching solutions (this one, `chokidar`, `nsfw`) is scaling well for VSCode for a long time and really want a single unifying solution for file watching in 1 library that is supporting all platforms VSCode supports.

The more pressure we get to change this, the better imho. So I would rather invest energy into that 1 library for all platforms than continuing what we do now.
